src/mpi/romio/adio/ad_daos/ad_daos_open.c

```diff
@@ -45,6 +45,21 @@ static int parse_filename(const char *path, char **_obj_name, char **_cont_name)
                 rc = ENOMEM;
                 goto out;
             }
+        } else {
+            char *new_dir = ADIOI_Calloc(strlen(cwd) + strlen(cont_name) + 1, sizeof(char));
+            if (new_dir == NULL) {
+                rc = -ENOMEM;
+                goto out;
+            }
+
+            strcpy(new_dir, cwd);
+            if (cont_name[0] == '.') {
+                strcat(new_dir, &cont_name[1]);
```

Comment:
What about `../a_file_name`?

Reply:
it's probably easier if i just use realpath(). let me do some testing and ask for reviews internally so i don't waste more of your time. thanks for your reviews.